api/src/main/java/org/jboss/marshalling/TraceInformation.java

```diff
@@ -59,17 +60,15 @@ private static TraceInformation getOrAddTraceInformation(Throwable t) {
         if (t == null) {
             throw new NullPointerException("t is null");
         }
-        Throwable c;
-        while (! (t instanceof TraceInformation)) {
-            c = t.getCause();
-            if (c == null) try {
-                t.initCause(c = new TraceInformation());
-            } catch (RuntimeException e) {
-                // ignored
-            }
-            t = c;
+        var optionalTi = Arrays.stream(t.getSuppressed()).filter(TraceInformation.class::isInstance).map(TraceInformation.class::cast).findFirst();
+        TraceInformation ti;
+        if (optionalTi.isEmpty()) {
+            ti = new TraceInformation();
+            t.addSuppressed(ti);
```

Comment:
Seems sensible. Did you notice if there was any test coverage for this facility?